ui/src/components/credentials/CredentialSubjectForm.tsx

```diff
@@ -0,0 +1,109 @@
+import { Number } from "src/components/credentials/attributes/Number";
+import { String } from "src/components/credentials/attributes/String";
+import { Attribute, ObjectAttribute } from "src/domain";
+
+function AnyAttribute({
+  attribute,
+  parents,
+}: {
+  attribute: Attribute;
+  parents: ObjectAttribute[];
+}): JSX.Element {
+  switch (attribute.type) {
+    case "boolean": {
+      return <Boolean attribute={attribute} parents={parents} />;
+    }
+    case "number":
+    case "integer": {
+      return <Number attribute={attribute} parents={parents} />;
+    }
+    case "string": {
+      return <String attribute={attribute} parents={parents} />;
+    }
+    case "null": {
+      return (
+        // ToDo: PID-587
+        <Typography.Text>Null attributes are not yet supported ({attribute.name})</Typography.Text>
+      );
+    }
+    case "multi": {
+      return (
+        // ToDo: PID-543
+        <Typography.Text>Multi attributes are not yet supported ({attribute.name})</Typography.Text>
+      );
+    }
+    case "array": {
+      return (
+        // ToDo: PID-589
+        <Typography.Text>Array attributes are not yet supported ({attribute.name})</Typography.Text>
+      );
+    }
+    case "object": {
+      return (
+        <CredentialSubjectForm
+          attributes={attribute.schema.properties || []}
+          parents={[...parents, attribute]}
+          space={false}
+        />
+      );
+    }
+  }
+}
+
+export function CredentialSubjectForm({
+  attributes,
+  parents = [],
+  space = true,
+}: {
+  attributes: Attribute[];
+  parents?: ObjectAttribute[];
+  space?: boolean;
+}) {
+  const shouldShowBreadcrumb = useRef<boolean>(true);
+  const form = [...attributes]
+    .sort((a, b) => (a.type !== "object" && b.type !== "object" ? 0 : a.type === "object" ? 1 : -1))
+    .map((attribute: Attribute) => {
+      const showBreadcrumb =
+        attribute.type !== "object" && parents.length > 1 && shouldShowBreadcrumb.current;
+
+      shouldShowBreadcrumb.current = !showBreadcrumb;
+
+      const attributeNode = showBreadcrumb ? (
+        <Space direction="vertical" size="middle">
+          <AttributeBreadcrumb parents={parents} />
+          <AnyAttribute attribute={attribute} parents={parents} />
+        </Space>
+      ) : (
+        <AnyAttribute attribute={attribute} parents={parents} />
+      );
+
+      const isRootAttribute = parents.length === 0;
+      const shouldShowTitle = isRootAttribute && attribute.type === "object";
+      const key = [...parents, attribute].map((parent) => parent.name).join(" > ");
+
+      return isRootAttribute ? (
+        <Card
+          className="ant-card-type-inner"
+          key={key}
+          title={shouldShowTitle ? attribute.schema.title || attribute.name : undefined}
+        >
+          {attributeNode}
+        </Card>
+      ) : (
+        <Fragment key={key}>{attributeNode}</Fragment>
+      );
+    });
+
+  return space ? (
```

Comment:
What does putting a single element in a `Space` do? It only affects the spacing between siblings.

Reply:
`form` is an array of components.

Reply:
Yep `form` is an array of the JSX elements of the cards. Without this space, it looks like this:

<img width="692" alt="image" src="https://user-images.githubusercontent.com/15896177/232744466-789fd5f9-016c-48a7-9db1-ad86db8231bf.png">

Reply:
Indeed, but my question is about [why we need the ternary](https://github.com/0xPolygonID/sh-id-platform/pull/297#discussion_r1169788850).

Reply:
Sorry I did not see your comment, I was adding to Adri's explanation.

We need the ternary because the form is recursive, and per the design, we only want to use cards (and therefore need the space) for the root elements (the first block in the image below).

If you take as an example the schema [Vaccination Certificate](https://schemas.serto.id/schema/vaccination-certificate) and log the `form` variable, you will see that the component `CredentialSubjectForm` is called 4 times, the first one is the one called from the outside (from `IssueCredentialForm`) but then, there are 3 more calls which occur recursively, from the component itself, to render the fields of the 3 objects present in the schema:

<img width="1023" alt="image" src="https://user-images.githubusercontent.com/15896177/232789155-37539643-2d70-417c-8b3e-6dbd18053070.png">
 
In the first call, we render the root elements with cards and therefore need the spaces. Later though, in the recursive calls, we don't want to space the input fields that are rendered inside the cards.

I have though realized that we don't need the prop `space` since we can reuse the variable `isRootAttribute` to infer the need for a space in the ternary. I've pushed this improvement.